module-apps/application-phonebook/windows/PhonebookNewContact.cpp

```diff
@@ -255,4 +260,30 @@ namespace gui
             }});
         application->switchWindow(gui::window::name::dialog_yes_no_icon_txt, std::move(metaData));
     }
+
+    void PhonebookNewContact::showContactDeletedNotification()
+    {
+        auto icon = "info_128px_W_G";
+        auto text = utils::translate("app_phonebook_contact_deleted");
+
+        auto metaData = std::make_unique<gui::DialogMetadataMessage>(gui::DialogMetadata{
+            contact->getFormattedName(ContactRecord::NameFormatType::Title), icon, text, "", [=]() -> bool {
+                std::unique_ptr<gui::SwitchData> data =
+                    std::make_unique<PhonebookItemData>(contact, newContactModel->getRequestType());
+                this->application->switchWindow(gui::name::window::main_window, std::move(data));
+
+                return true;
+            }});
+        application->switchWindow(gui::window::name::dialog_confirm, std::move(metaData));
+    }
+
+    bool PhonebookNewContact::checkIfContactWasDeletedDuringEditProcess()
+    {
+        auto contactByID = DBServiceAPI::ContactGetByID(application, contact->ID);
+
+        return contactByID->size() == 1 and contactByID->front().ID == 0 and
+               contactByID->front().primaryName.empty() and contactByID->front().alternativeName.empty() and
+               contactByID->front().numbers.empty();
```

Comment:
Are these all comparisons necessary? Isn't it enough to check if `contactByID->empty()`?

Reply:
Unfortunately not, for some reasons if the contacts do not exist in a database a vector with single empy ContactRecord is returned. Thats why I created all these checks